ambry-cloud/src/main/java/com/github/ambry/cloud/azure/StorageClient.java

```diff
@@ -382,6 +410,22 @@ protected BlobServiceAsyncClient createBlobStorageAsyncClient() {
     }
   }
 
+  protected BlobServiceClient createBlobStorageSyncClient() {
```

Comment:
Why do we need sync client? Is this for testing purpose?

Reply:
This is not for testing. V2 will use syncClient. There are several issues with the AsyncClient.

1. Async is a good principle in general but we need to evaluate it in each context before applying it. AsyncClient offers absolutely ZERO benefits for V2. V1 is a readability and debugging nightmare because of AsyncClient. Please read the V1 code and/or talk to Arun. 
2. AsyncClient was implemented in V1 because VCR was expected to server customer traffic. That is simply not the case anymore. 
3. Replication does not proceed to fetch a blob from server without copying the previous one out to Azure. So AsyncClient simply doesn't make sense.
4. AsyncClient was implemented to override some of the handlers post-completion and do some operations on cosmosDB including storing metadata. Not the case anymore. 
5. SyncClient is a wrapper over AsyncClient in the library. They complete the future returned by AsyncClient to implement the syncClient. 
6. I am willing to sacrifice any performance improvements that AsyncClient may offer for the sake of readability, speed of development and ease of debugging that syncClient offers. 
7. We can revisit this if the performance of syncClient is unacceptable.

Reply:
@Arun-LinkedIn Arun, do you have any concern here? I know we have put quite some effort to change the cloud implement from sync mode to async way.